drv/cpu-seq-api/src/lib.rs

```diff
@@ -31,6 +32,22 @@ pub enum SeqError {
     ServerRestarted,
 }
 
+#[derive(Copy, Clone, Debug, FromPrimitive, Eq, PartialEq, AsBytes, Count)]
+#[repr(u8)]
+pub enum StateChangeReason {
+    /// TThe system has just received power, so the sequencer has booted the
+    /// host CPU.
+    InitialPowerOn = 1,
+    /// A power state change was requested by the control plane.
+    ControlPlane,
+    /// The host OS requested that the system power off without rebooting.
+    HostPowerOff,
+    /// The host OS requested that the system reboot.
+    HostReboot,
```

Comment:
I also considered making these one thing, like
```rust
HostPowerOff { reboot: bool }
```
but that makes this enum two bytes, and probably also embiggens the ringbuf entry...

Reply:
I think you might get the desired effect with another enum; `HostPowerOff(PowerOffStyle)`

```rs
enum PowerOffStyle {
  Shutdown,
  Reboot
}
```

As I understand, as long as there aren't multiple enum variants with a value, and the only value is a non-repr enum then Rust will smash the two together. At least, that's what I've seen happening in one place in code I work with.

Reply:
Huh, I didn't realize Rust would do that when the outermost enum has a `repr` attribute. That's kinda nice! I'll look into it, thanks!

Reply:
Honestly, though, I'm not sure if that's really any nicer than the current thing, in this case --- the only reason I would've done it with a `bool` is so that the caller that uses this variant, the `power_off_host` function in `task-host-sp-comms`, could just pass its `reboot` field rather than having to say `if reboot { ... }`:
https://github.com/oxidecomputer/hubris/blob/63995e0cdd0e5087244bbd413db7d00b6654bfc5/task/host-sp-comms/src/main.rs#L376-L386

But, if we add another enum here, it still has to do that, so. I'm not sure if this is worth changing.

Reply:
I was really confused and surprised when I found out as well, but it's pretty nice at times.

You could implement a `From<bool>` for the inner enum to hide the if-else away. That being said, it's arguably not that much nicer.

Reply:
Given that a `From<bool>` implementation would only be used in one place (`power_off_host` in `task-host-sp-comms`) but would have to be defined in `cpu-seq-api`, I think I'll leave it as-is. I think it's easier to understand when the conversion from the `reboot: bool` to a `StateChangeReason` occurs in the one place where it actually happens, instead of hiding it behind a `From` impl in a separate crate that isn't used anywhere else.

Thanks for the suggestions, though!